packages/clarity-js/src/layout/dom.ts

```diff
@@ -117,31 +117,28 @@ export function add(node: Node, parent: Node, data: NodeInfo, source: Source): v
 export function update(node: Node, parent: Node, data: NodeInfo, source: Source): void {
     let id = getId(node);
     let parentId = parent ? getId(parent) : null;
-    let nextId = getNextId(node);
+    let previousId = getPreviousId(node);
     let changed = false;
 
     if (id in values) {
         let value = values[id];
         value.metadata.active = true;
 
         // Handle case where internal ordering may have changed
-        if (value["next"] !== nextId) {
+        if (value.previous !== previousId) {
             changed = true;
-            value["next"] = nextId;
+            value.previous = previousId;
         }
 
         // Handle case where parent might have been updated
-        if (value["parent"] !== parentId) {
+        if (value.parent !== parentId) {
             changed = true;
-            let oldParentId = value["parent"];
-            value["parent"] = parentId;
+            let oldParentId = value.parent;
+            value.parent = parentId;
             // Move this node to the right location under new parent
             if (parentId !== null && parentId >= 0) {
-                if (nextId !== null && nextId >= 0) {
-                    values[parentId].children.splice(nextId + 1, 0 , id);
-                } else {
-                    values[parentId].children.push(id);
-                }
+                let childIndex = previousId !== null ? values[parentId].children.indexOf(previousId) + 1 : values[parentId].children.length;
```

Comment:
>!== [](start = 44, length = 3)

Style question: Do you prefer to write these as above or as: 

previousId === null ? values[values[parentId].children.length] : values[parentId].children.indexOf(previousId) + 1 

So that the condition is not negated? To me the above is my preference as I find it easier to understand but tastes on this point vary.

Reply:
Your recommendation looks good, made the change.